src/ServiceControl.Persistence.InMemory/InMemoryCustomCheckDataStore.cs

```diff
@@ -10,8 +11,47 @@ namespace ServiceControl.Persistence.InMemory
 
     class InMemoryCustomCheckDataStore : ICustomChecksDataStore
     {
-        public Task<CheckStateChange> UpdateCustomCheckStatus(CustomCheckDetail detail) => throw new System.NotImplementedException();
-        public Task<QueryResult<IList<CustomCheck>>> GetStats(PagingInfo paging, string status = null) => throw new System.NotImplementedException();
         public Task Setup() => Task.CompletedTask;
+
+        public Task<CheckStateChange> UpdateCustomCheckStatus(CustomCheckDetail detail)
+        {
+            if (storage.ContainsKey(detail.CustomCheckId))
+            {
+                var storedCheck = storage[detail.CustomCheckId];
+                if (storedCheck.HasFailed == detail.HasFailed)
+                {
+                    return Task.FromResult(CheckStateChange.Unchanged);
+                }
+            }
```

Comment:
Should this update the state here?

Reply:
There is a bit of code that maps from `CustomCheckDetail` that does this exact thing in other persistences. That potentially can be refactored into something reusable and removed from individual persistence layers. I'll update that in a separate PR.